File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    return arr1.map((v, i) => (+v.toString(2) + +arr2[i].toString(2)).toString())
            .map((v) => v.length < n ? '0'.repeat(n - v.length) + v : v) // 부족한 자리수만큼 앞에 0 붙이기
            .map((v) => v.split('').map((v) => v === '0' ? ' ' : '#').join('')) // 0이면 ' ', 아니면 '#'으로 매핑하기  ["0","0","2","2","0"]
    
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i in range(len(arr1)):
        a1 = bin(int(arr1[i]))[2:].zfill(n) # 2진수 변경 -> 접두사 자름 -> n자리 맞춤
        a2 = bin(int(arr2[i]))[2:].zfill(n)
        
        line = ''
        for j in range(len(a1)):
            if int(a1[j]) + int(a2[j]) == 0:
                line += ' '
            else:
                line += '#'
        answer.append(line)
    return answer